www_src/components/color-spectrum/color-spectrum.jsx

```diff
@@ -1,20 +1,116 @@
+var Color = require('color');
 
-// Doesn't do much yet, but renders the hue and saturation spectrum
+var HSB = React.createClass({
+  getBackgroundColor: function () {
+    var color = this.props.color.clone();
+    color.hsv(color.hue(), 100, 100);
+    return color.rgbaString();
+  },
+  getSBPosition: function () {
+    var hsv = this.props.color.hsv();
+    return {
+      top: `${100 - hsv.v}%`,
+      left: `${hsv.s}%`
+    };
+  },
+  changeHue: function (e) {
+    var color = this.props.color;
+    color.hue(e.target.value);
+    this.props.onChange(color);
+  },
+  changeSB: function (e) {
+    var color = this.props.color;
 
-var ColorSpectrum = React.createClass({
+    var x = e.touches[0].clientX - this.bounding.left;
+    var y = e.touches[0].clientY - this.bounding.top;
+
+    color.hsv(color.hue(), x / this.bounding.width * 100, 100 - y / this.bounding.height * 100);
+    this.props.onChange(color);
+  },
+  onTouchStart: function(e) {
+    e.preventDefault();
+    this.bounding = this.refs.sl.getDOMNode().getBoundingClientRect();
+    this.changeSB(e);
+  },
+  onTouchMove: function (e) {
+    this.changeSB(e);
+  },
   render: function () {
+
     return (<div className="spectrum-container">
       <div className="spectrum-left">
-        <div className="saturation" style={{backgroundColor: this.props.value}}>
+        <div ref="sl" className="saturation" style={{backgroundColor: this.getBackgroundColor()}} onTouchStart={this.onTouchStart} onTouchMove={this.onTouchMove}>
           <div className="saturation-white" />
           <div className="saturation-black" />
+          <div ref="marker" className="sl-marker" style={this.getSBPosition()}/>
         </div>
-        <input className="hue" type="range" orient="vertical" />
+        <input value={this.props.color.hue()} min={0} max={359} className="hue" type="range" orient="vertical" onChange={this.changeHue} />
       </div>
       <div className="spectrum-right" />
     </div>);
   }
 });
 
+var Alpha = React.createClass({
+  onChangeAlpha: function (alpha) {
+    var color = this.props.color;
+    color.alpha(alpha);
+    this.props.onChange(color);
+  },
+  render: function () {
+    return (<div>
+      <Slider value={this.props.color.alpha()} max={1} step={0.01} onChange={this.onChangeAlpha} />
+    </div>);
+  }
+});
+
+var RGB = React.createClass({
+  generateOnChangeCallback: function (label) {
+    return (value) => {
+      var color = this.props.color;
+      color[label](value);
+      this.props.onChange(color);
+    };
+  },
+  render: function () {
+    return (<div>
+      {['Red', 'Green', 'Blue'].map((label, i) => {
+        var lowercaseLabel = label.toLowerCase();
+        var currentValue = this.props.color[lowercaseLabel]();
+        return (<div key={label} className="form-group">
+          <label>{label}</label>
+          <Slider value={currentValue} max={255} onChange={this.generateOnChangeCallback(lowercaseLabel)} />
```

Comment:
in part "I don't trust browsers to respect defaults", but mostly it's a matter of "let's not leave implicit what we can be explicit about,so that other devs/contributors reading the code immediately get it"